src/main/java/gov/nasa/pds/harvest/util/xml/XmlIs.java

```diff
@@ -0,0 +1,69 @@
+package gov.nasa.pds.harvest.util.xml;
+
+import java.io.FileReader;
+import javax.xml.transform.Source;
+import javax.xml.transform.sax.SAXSource;
+import org.xml.sax.InputSource;
+import net.sf.saxon.Configuration;
+import net.sf.saxon.lib.ParseOptions;
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.om.TreeInfo;
+import net.sf.saxon.xpath.XPathEvaluator;
+
+public class XmlIs {
+  private static String ANY_TYPE = "logical_identifier";
+  private static boolean interogate(String filename, String type) {
+    if (filename.endsWith (".xml") || filename.endsWith (".lblx")) {
+      try {
+        XPathEvaluator xpath = new XPathEvaluator();
+        Configuration configuration = xpath.getConfiguration();
+        configuration.setLineNumbering(true);
+        configuration.setXIncludeAware(false);
+        ParseOptions options = new ParseOptions();
+        options.withLineNumbering(true);
+        options.withXIncludeAware(false);
+        Source source = new SAXSource(new InputSource(new FileReader(filename)));
+        TreeInfo docInfo = configuration.buildDocumentTree(source , options);
+        NodeInfo ia=null,lid=null,pcls=null;
+        for (NodeInfo top : docInfo.getRootNode().children()) {
+          for (NodeInfo child : top.children()) {
+            if ("Identification_Area".equals(child.getLocalPart())) {
+              if (ia == null) {
+                ia = child;
```

Comment:
@al-niessner do we want to break out of here once we find the identification area in the event this is a huge label? if there are 2 identification areas, there are bigger problems, and it isn't responsibility of Harvest to validate that is true.

Reply:
No. Pluto might have an XML as data with multiple Identification_Area and this helps us say, not a PDS4.

Reply:
@al-niessner I would almost prefer the software just breaks down on bad data that we assume is right, versus assuming it all could be wrong, but not worth the time right now.